### DISCUSSION TITLE
Is there a feature to trigger script after connect  or disconnect to xrdp?

### DISCUSSION THREAD

**Discussion by jjasoncool** [neutrinolabs/xrdp#2681] [Q&A] [answered]
I need to use `x11vnc` and `xrandr` to handle screen resolution and position. 

However, it seems that there is no way to configure it to adjust the resolution and screen condition with a script after connecting or disconnecting with xrdp. 

I would like to ask how to implement this and if there are any recommended methods?

Will it be added as a new feature in the future?

Thank you.

**Comment by matt335672** [comment, marked as answer]
Existing functionality in xrdp v0.9.x allows a script to be run when a client reconnects (not connects for the first time). Have a look at `ReconnectScript` in `/etc/xrdp/sesman.ini`.

There is currently no disconnect script, but this has been requested. See #2090 and #1598. We've just finished changing the sesman architecture in development builds which will make this possible in the future.

**Reply by jjasoncool** [reply]
Thank you. I hope this can be written in document.

**Reply by jjasoncool** [reply]
because I am newbie, so I have google that and try to ask chatGPT, but can't get good answer.
Until you answer this.

**Reply by matt335672** [reply]
It's documented in the [manpage for sesman.ini](https://manpages.ubuntu.com/manpages/focal/man5/sesman.ini.5.html). At the moment there's no other documentation I'm aware of.

Is there something else you need to know?

**Reply by jjasoncool** [reply]
Currently, there are no other questions. If there are any problems with finding other documents, I will post a new discussion here. Thank you.

**Reply by jjasoncool** [reply]
> It's documented in the [manpage for sesman.ini](https://manpages.ubuntu.com/manpages/focal/man5/sesman.ini.5.html). At the moment there's no other documentation I'm aware of.
> 
> Is there something else you need to know?

ReconnectScript doesn't work on any-vnc connection. (I have try reconnect to xrdp, not first time.)

**Reply by matt335672** [reply]
The reconnect script needs sesman to be involved. In order to get this, you need start a session via xrdp, and not simply use it as a proxy.

If you're just using xrdp as a proxy, you'll need to detect the re-connection on the Xvnc server itself. There are a few reasons why it is like this. An important one is that on some platforms the xrdp service has no privileges to run the script as another user.
